### DISCUSSION TITLE
there is any props to set size or max height of years selection ?

### DISCUSSION THREAD

**Discussion by GeuntaBuwono** [gpbl/react-day-picker#2376] [Support] [answered]
since there is size props on html #https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/size 
what props for this ?

**Comment by gpbl** [comment]
@GeuntaBuwono if you mean the height of the dropdown when opened

<img width="808" alt="Screenshot 2024-08-17 at 5 30 31 AM" src="https://github.com/user-attachments/assets/47ead1ad-cc39-4846-8fde-8c076a2fee39">

than this depends from the behavior of the Operating System. 

While I believe this is not a real issue, you have different options here:

1. implement a custom select component to replace the native dropdown:

```tsx
<DayPicker components={{ Select: MyCustomSelect, Option: MyCustomOption }} />
```

where `MyCustomSelect` and `MyCustomOption` are components replacing the native `select` and `option` HTML elements. These could implement their own user interface, like a https://headlessui.com/v1/react/listbox

2. do not use `dropdown` as `captionLayout`. If there are so many years to choose from, consider adding a different input field to control the year or the displayed month. This will improve usability of your user interface. See https://daypicker.dev/docs/navigation#onmonthchange

Since this is not the first time we get this request, I'll see to [add more examples](https://github.com/gpbl/react-day-picker/issues/2378) in the docs and see if you could add different ways to navigate between years (any idea?).

**Comment by GeuntaBuwono** [comment]
@gpbl I've tried the solution. but in my case is for DoB 

```
startMonth={new Date("1900-01-01")}
endMonth={new Date("2000-01-01")}
hidden={{
before: new Date("1900-01-01"),
after: new Date("2000-01-01"),
}}
```

![image](https://github.com/user-attachments/assets/8b3bb505-b9d7-4acc-af57-d3e6b1ab34cf)

**Reply by gpbl** [reply, marked as answer]
@GeuntaBuwono this is the standard behavior of MacOS. I'm not aware about any way to change it, and I don't see even why. If you need to improve the user experience, do not use a drop-down and ask users to enter the year via an input field instead of a select.
